akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -671,4 +671,20 @@ object Source {
   def unfoldResourceAsync[T, S](create: () ⇒ Future[S], read: (S) ⇒ Future[Option[T]], close: (S) ⇒ Future[Done]): Source[T, NotUsed] =
     Source.fromGraph(new UnfoldResourceSourceAsync(create, read, close))
 
+  /**
+   * Materializes into a sink connected to a source, i.e. the sink pushes into the source:
+   *
+   * +----------+       +----------+
+   * >   Sink   |------>|  Source  >
+   * +----------+       +----------+
+   *
+   * Should be provided by Akka Streams, see https://github.com/akka/akka/issues/24853.
+   */
+  def sinkToSource[M]: RunnableGraph[(Sink[M, NotUsed], Source[M, NotUsed])] =
```

Comment:
Why add this method to `Sink`? It creates both a sink and source on materialization. I'd rather put it into `RunnableGraph`.

Reply:
See my comment 
https://github.com/akka/akka/pull/25150#issuecomment-393102788

Reply:
RunnableGraph would perhaps make sense but it would be completely undiscoverable as there are no other factories (`fromGraph` is the only one) there.